frontend/controller/actions/group.js

```diff
@@ -678,12 +678,18 @@ export default (sbp('sbp/selectors/register', {
 
     await sbp('gi.actions/group/joinChatRoom', {
       ...omit(params, ['options', 'data', 'hooks']),
-      data: {
-        chatRoomID
-      },
+      data: { chatRoomID },
       hooks: {
         onprocessed: (msg) => {
-          sbp('state/vuex/commit', 'setCurrentChatRoomId', { chatRoomId: chatRoomID, groupId: msg.contractID() })
+          const fnEventHandled = (cID, message) => {
+            if (cID === chatRoomID) {
+              if (sbp('state/vuex/getters').isJoinedChatRoom(chatRoomID)) {
+                sbp('state/vuex/commit', 'setCurrentChatRoomId', { chatRoomID, groupID: msg.contractID() })
+                sbp('okTurtles.events/off', EVENT_HANDLED, fnEventHandled)
+              }
+            }
+          }
+          sbp('okTurtles.events/on', EVENT_HANDLED, fnEventHandled)
```

Comment:
The `onprocessed` hook is already designed to be called on `EVENT_HANDLED` after the message is received back and processed.

So this unnecessary handler here needs to be removed.

Reply:
Here, we are not waiting for the current event (`gi.contracts/group/joinChatRoom`) to be finished. The function `fnEventHandled` is to wait for `gi.contracts/chatroom/join` event to be finished, and the user is fully joined chatroom.

Reply:
This seems like a very strange place to do this then.

Wouldn't it make more sense to do this in the `sideEffect` of `gi.contracts/chatroom/join`?

Reply:
I don't think we should do that way because it will cause the issue #1960.

Reply:
I see, in that case, since this looks kind of confusing and weird, could you please add a comment? Something along these lines (right above line 683):

```js
// joinChatRoom sideEffect will trigger a call to 'gi.actions/chatroom/join', we want
// to wait for that action to be received and processed, and then switch the UI to the
// new chatroom. We do this here instead of in the sideEffect for chatroom/join to
// avoid causing the UI to change in other open tabs/windows, as per bug:
// https://github.com/okTurtles/group-income/issues/1960
```